master/internal/rm/agentrm/resource_pool.go

```diff
@@ -476,6 +477,7 @@ func (rp *resourcePool) allocateResources(req *sproto.AllocateRequest) bool {
 		ResourcePool:      rp.config.PoolName,
 		Resources:         sprotoResources,
 		JobSubmissionTime: req.JobSubmissionTime,
+		RequestTime:       req.RequestTime,
```

Comment:
If I start an experiment, wait till it starts

then restart master we will write multiple queued timed rows into task stats


```
determined=# SELECT * FROM task_stats;
              allocation_id               | event_type |          start_time           |           end_time            | container_id 
------------------------------------------+------------+-------------------------------+-------------------------------+--------------
 1.153b99d5-cca4-4334-b650-422b2f3354b9.1 | QUEUED     | 2024-01-31 13:33:01.351726+00 | 2024-01-31 13:33:01.530818+00 | 
 1.153b99d5-cca4-4334-b650-422b2f3354b9.1 | QUEUED     | 2024-01-31 13:33:17.500608+00 | 2024-01-31 13:33:17.506029+00 | 
(2 rows)
```

This feels strange to me. I'm not sure if this would cause any issues but it feels like a mistake more than actual intended behavior.

Reply:
Correct me if I've misunderstood. Based on the code, it seems like the intention is to record both the start queued time and the end queued time for each allocation. In this case, it means the allocation was in the queue twice? The second time was when the experiment was restored, sproto.AllocateRequest was [created again](https://github.com/determined-ai/determined/blob/main/master/internal/trial.go#L402) and the [resource was allocated](https://github.com/determined-ai/determined/blob/main/master/internal/task/allocation.go#L625
).

Reply:
I don't think we support an allocation being queued in multiple times. I think there is an assumption we create and run an allocation, then make complete and make one if it needs to be.

I guess it being recorded as queue twice is kinda misleading since it was only ever queued and run once.

Do you see the same results in the db if you run an experiment, wait till it is running, then restart?

Reply:
I see the same result.

Reply:
[The current code](https://github.com/determined-ai/determined/blob/main/master/internal/trial.go#L766-L769) finds allocations without end_time in the database and try to restores them. So, just to clarify, are we supposed to create a new allocation when it is restored?

Reply:
No I think the issue is on the queued time side. We use the same allocation but we should only ever go from "queued" to "running" once. Restoring won't cause an allocation to back in the queue ever.